autoreadme.go

```diff
@@ -159,17 +163,18 @@ func getDoc(dir string) (Doc, error) {
 		return Doc{}, err
 	}
 
+	goFiles := map[string]struct{}{}
+	for _, fs := range [][]string{bi.GoFiles, bi.CgoFiles, bi.TestGoFiles} {
```

Comment:
Examples can (and generally should!) be in `bi.XTestGoFiles`. 

Further, there's not much gained by trying to parse everything at once. It muddles the steps taken by trying to do much at once. It's going to be clearer to parse the test files separately using `ParseFile`.